plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/action/top_queries/TopQueriesResponse.java

```diff
@@ -0,0 +1,115 @@
+import java.util.stream.Collectors;
+
+/**
+ * Transport response for cluster/node level top queries information.
+ *
+ * @opensearch.internal
+ */
+@PublicApi(since = "1.0.0")
+public class TopQueriesResponse extends BaseNodesResponse<TopQueries> implements ToXContentFragment {
+
+    private static final String CLUSTER_LEVEL_RESULTS_KEY = "top_queries";
+
+    public TopQueriesResponse(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    public TopQueriesResponse(ClusterName clusterName, List<TopQueries> nodes, List<FailedNodeException> failures) {
+        super(clusterName, nodes, failures);
+    }
+
+    @Override
+    protected List<TopQueries> readNodesFrom(StreamInput in) throws IOException {
+        return in.readList(TopQueries::new);
+    }
+
+    @Override
+    protected void writeNodesTo(StreamOutput out, List<TopQueries> nodes) throws IOException {
+        out.writeList(nodes);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        List<TopQueries> results = getNodes();
+        builder.startObject();
+        toClusterLevelResult(builder, params, results);
+        return builder.endObject();
+    }
+
+    @Override
+    public String toString() {
+        try {
+            XContentBuilder builder = XContentFactory.jsonBuilder().prettyPrint();
+            builder.startObject();
+            this.toXContent(builder, EMPTY_PARAMS);
+            builder.endObject();
+            return builder.toString();
+        } catch (IOException e) {
+            return "{ \"error\" : \"" + e.getMessage() + "\"}";
+        }
+    }
+
+    /**
+     * Merge top n queries results from nodes into cluster level results in XContent format.
+     *
+     * @param builder XContent builder
+     * @param params serialization parameters
+     * @param results top queries results from all nodes
+     * @throws IOException if an error occurs
+     */
+    private void toClusterLevelResult(XContentBuilder builder, Params params, List<TopQueries> results) throws IOException {
+        List<SearchQueryLatencyRecord> all_records = results.stream()
+            .map(TopQueries::getLatencyRecords)
+            .flatMap(Collection::stream)
+            .sorted(Collections.reverseOrder())
+            .collect(Collectors.toList());
+        builder.startArray(CLUSTER_LEVEL_RESULTS_KEY);
+        for (SearchQueryLatencyRecord record : all_records) {
+            record.toXContent(builder, params);
+        }
+        builder.endArray();
+    }
+
+    /**
+     * build node level top n queries results in XContent format.
+     *
+     * @param builder XContent builder
+     * @param params serialization parameters
+     * @param results top queries results from all nodes
+     * @throws IOException if an error occurs
+     */
+    private void toNodeLevelResult(XContentBuilder builder, Params params, List<TopQueries> results) throws IOException {
```

Comment:
private method not used? Are we planning to use this in the future

Reply:
Removing this one since we only plan to show cluster level results.